pilot/pkg/features/pilot.go

```diff
@@ -226,6 +226,14 @@ var (
 			"if headless services have a large number of pods.",
 	)
 
+	EnableHeadlessServicePassthrough = env.RegisterBoolVar(
+		"PILOT_ENABLE_HEADLESS_SERVICE_PASSTHROUGH",
+		true,
+		"If enabled, for a headless service/stateful set in Kubernetes, pilot will not "+
+			"send endpoints over EDS, but instead will set up passthrough resolution. This feature should be disabled "+
+			"if sidecars need to access headless services via a HTTP proxy egress listener.",
+	)
```

Comment:
change to PILOT_ENABLE_EDS_FOR_HEADLESS_SERVICES (defaults to false) (and change the variable name/explanation  accordingly).